internal/config/profile_integration_test.go

```diff
@@ -0,0 +1,279 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build integration
+
+package config
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	atlasProfile   = "atlas"
+	newProfileName = "newProfileName"
+)
+
+func createProfile(profileContents string) error {
+	p.fs = afero.NewMemMapFs()
+
+	testConfigDir, err := configHome()
+	if err != nil {
+		return err
+	}
+
+	filename := fmt.Sprintf("%v/%v.toml", testConfigDir, ToolName)
+
+	err = afero.WriteFile(p.fs, filename, []byte(profileContents), 0600)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createProfileWithOneNonDefaultUser() error {
+	const contents = `
+[atlas]
+  base_url = "http://base_url.com/"
+  ops_manager_skip_verify = "false"
+  org_id = "5cac6a2179358edabd12b572"
+`
+	return createProfile(contents)
+}
+
+func createProfileWithOneDefaultUserOneNonDefault() error {
+	const contents = `
+[default]
+  base_url = "http://base_url.com/"
+  ops_manager_skip_verify = "false"
+  org_id = "5cac6a2179358edabd12b572"
+  profile_id = "5cac6a2179358edabd12b572"
+
+[atlas]
+  base_url = "http://atlas.com/"
+  ops_manager_skip_verify = "false"
+  org_id = "5cac6a2179358edabd12b572"
+`
+	return createProfile(contents)
+}
+
+func createProfileWithFullDescription() error {
+	const contents = `
+[default]
+  base_url = "http://base_url.com/"
+  ops_manager_url = "http://om_url.com/"
+  ops_manager_ca_certificate = "/path/to/certificate"
+  ops_manager_skip_verify = "false"
+  public_api_key = "some_public_key"
+  private_api_key = "some_private_key"
+  org_id = "5cac6a2179358edabd12b572"
+  profile_id = "5cac6a2179358edabd12b572"
+  service = "cloud"
+`
+	return createProfile(contents)
+}
+
+func TestProfile_List(t *testing.T) {
+	err := createProfileWithOneDefaultUserOneNonDefault()
+	if err != nil {
+		t.Error(err)
+	}
+
+	if err := Load(); err != nil {
+		t.Error(err)
+	}
+
+	availableProfiles := List()
+	assert.Len(t, availableProfiles, 2, "expected to find 2 profiles")
+}
+
+func TestProfile_Get_FullProfile(t *testing.T) {
+	err := createProfileWithFullDescription()
+	if err != nil {
+		t.Error(err)
+	}
+
+	SetName(DefaultProfile)
+
+	if err := Load(); err != nil {
+		t.Error(err)
+	}
+
+	desc := Get()
+
+	a := assert.New(t)
+	a.Equal("default", Name())
+	a.Len(desc, 9)
+	a.Equal("cloud", Service())
+	a.Equal("some_public_key", PublicAPIKey())
+	a.Equal("some_private_key", PrivateAPIKey())
+	a.Equal("http://om_url.com/", OpsManagerURL())
+	a.Equal("/path/to/certificate", OpsManagerCACertificate())
+	a.Equal("false", OpsManagerSkipVerify())
+}
+
+func TestProfile_Get_Default(t *testing.T) {
+	err := createProfileWithOneDefaultUserOneNonDefault()
+	if err != nil {
+		t.Error(err)
```

Comment:
same and multiple of other test, this is how I like to think about it, can my test continue after this error? if so let's use `t.Error` if not (I'll just keep getting errors) let's use `t.Fatal`

Reply:
I think for me I'd rather use `Fatal` (practically) everywhere as it saves thinking about possible failure conditions on tests, when the test has probably lost any meaningful value as soon as the first error is hit. (It's why I kept using `Error` by default because it doesn't feel intuitive that a test should continue running after reaching an error)

Reply:
that works for me as well, keep in mind that doing asser.Equal/True/Len multiple times has a similar behaviour to doing `t.Error` so it's quite the "go way"